contracts/duckies/Garden.sol

```diff
@@ -23,33 +25,32 @@ contract Garden is Initializable, AccessControlUpgradeable, UUPSUpgradeable {
 
 	// Constants
 	uint8 public constant REFERRAL_MAX_DEPTH = 5;
-	uint256 internal _DUCKIES_SUPPLY_CAP;
-	uint8 internal constant _MAX_HALVING_STEP = 5;
 	uint8 internal constant _REFERRAL_PAYOUT_DIVIDER = 100;
 
+	enum BountyType {
+		TransferBountyToken, // encodedData: abi.encode(address, uint256, uint8[5]) - encoded token address, amount to pay, referrer payouts
+		MintDucklingsNFT // encodedData: 0x (empty). May be extended in the future.
+	}
+
 	// Bounty Message for signature verification
 	struct Bounty {
-		uint256 amount;
-		address tokenAddress;
+		BountyType type_;
+		bytes encodedData; // specific to BountyType
```

Comment:
I like the flexibility, but, could we read data into a structure, for example we call it
```
struct PayoutParams {
    address token;
    uint256  amount;
    uint8      payouts;
}

struct FactoryParams {
    address token;        // NFT contract address
    uint256  quantity;    // Size of card pack
    uint8      collection;  // index of collection
}

struct MeldParams {
    bytes32[5]  family;  // Array of DNA of the cards, or TokenID, need to verify ownership
}

// Then in the switch case we unpack into
// memory MeldParams args = unpack(data);
// meldCards(args);
```

Reply:
Sure, of course we can.